.github/workflows/nightly-build-publish.yml

```diff
@@ -1,32 +1,106 @@
-name: Nightly Build and Publish
+name: Nightly Test and Publish
 
 on:
-#  schedule:
-#    - cron: "0 2 * * *"
   workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version (e.g. 0.2.21-dev.20250911)'
+        required: false
+        type: string
+      commit_id:
+        description: 'Commit ID for "cut" (default: origin/main)'
+        required: false
+        type: string
+      inference_provider:
+        description: 'Inference provider (fireworks, together)'
+        required: false
+        type: string
+        default: 'fireworks'
+      cut_mode:
+        description: 'Mode to run the script in (test-and-cut, cut-only)'
+        type: string
+        default: 'test-and-cut'
+      llama_stack_only:
+        description: 'Only cut a llama-stack nightly release'
+        required: false
+        type: boolean
+        default: false
+#  schedule:
+#    - cron: "0 4 * * *"  # Run every day at 4 AM UTC
 
 jobs:
-  nightly:
+  generate-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+    - uses: actions/checkout@v4
+    - name: Generate nightly version
+      id: version
+      run: |
+        if [ -n "${{ inputs.version }}" ]; then
+          echo "Using provided VERSION: ${{ inputs.version }}"
+          echo "version=${{ inputs.version }}" >> $GITHUB_OUTPUT
+        else
+          echo "Extracting base version from llama-stack repo..."
+
+          # Clone llama-stack repo to extract version
+          git clone --depth 1 "https://x-access-token:${{ secrets.LLAMA_REPOS_PAT }}@github.com/meta-llama/llama-stack.git" temp-version-check
+          cd temp-version-check
+
+          # Extract version from pyproject.toml
+          base_version=$(grep '^version = ' pyproject.toml | sed 's/version = "\(.*\)"/\1/')
```

Comment:
this kind of version manipulation has bitten me in the past. i think you can use a tool like `tomli` to programmatically extract this.